src/extension/ip_extraction.c

```diff
@@ -463,14 +466,15 @@ static bool _parse_ip_address_maybe_port_pair(
         if (!pos_close) {
             return false;
         }
-        return _parse_ip_address(addr + 1, pos_close - (addr + 1), out);
+        return _parse_ip_address(
+            addr + 1, pos_close - (addr + 1), ip_or_error, out);
     }
     const char *colon = memchr(addr, ':', addr_len);
-    if (colon) {
-        return _parse_ip_address(addr, colon - addr, out);
+    if (colon && zend_memrchr(addr, ':', addr_len) == colon) {
```

Comment:
You don't seem to add any new test case for when there is more than one colon character.

Reply:
If `addr` does not have `[` bracket and it has two columns, then it is meant to be a v6 ip without port. Given that, it will then let `_parse_ip_address` parse the whole len of addr and determine wether or not it's a valid IP. 
In the case of ipv4 with port, it can only have one column.
Both cases are tested I think.